model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py

```
@@ -0,0 +1,646 @@
+                nbits = self._get_weight_nbits(n, attr, bitwidth_mode, qc)
+                bytes_ = size * nbits / 8
+            attr_util[attr] = Utilization(size, bytes_)
+
+        total_weights = sum(attr_util.values())
+        return total_weights, attr_util
+
+    def compute_activations_utilization(self,
+                                        target_criterion: TargetInclusionCriterion,
+                                        bitwidth_mode: BitwidthMode,
+                                        act_qcs: Optional[Dict[BaseNode, NodeActivationQuantizationConfig]] = None):
+        return self.compute_cut_activation_utilization(target_criterion, bitwidth_mode, act_qcs)
+
+    def compute_cut_activation_utilization(self,
+                                           target_criterion: TargetInclusionCriterion,
+                                           bitwidth_mode: BitwidthMode,
+                                           act_qcs: Optional[Dict[BaseNode, NodeActivationQuantizationConfig]]) \
+            -> Tuple[float, Dict[Cut, Utilization], Dict[Cut, Dict[BaseNode, Utilization]]]:
+        """
+        Calculate graph activation cuts utilization.
+
+        Args:
+            target_criterion: criterion to include weights for computation.
+            bitwidth_mode: bit-width mode for the computation.
+            act_qcs: custom configuration for the custom bit mode. Must provide configuration for all configurable
+              activations.
+
+        Returns:
+            - Total utilization.
+            - Total utilization per cut.
+            - Detailed utilization per cut per node.
+        """
+        if target_criterion != TargetInclusionCriterion.AnyQuantized:    # pragma: no cover
+            raise NotImplementedError('Computing MaxCut activation utilization is currently only supported for quantized targets.')
+
+        graph_target_nodes = self._get_target_activation_nodes(target_criterion, include_reused=True)
```

what case is it handling? (I recall that you said something about this regarding one of the tests but I don't remember exactly what was the issue).
Anyway, if this is test-related and can be handled in the tests, I think that in such a case we should throw an error don't you think? returning 0, {}, {} is cryptic a defiantly not indicating the activation utilization correctly (this is a case of incompatible input to the function if I understand correctly)

---

If the graph does not contain the nodes per target_criterion, there is no point to proceed to computing the cuts. 
The question is whether to consider 0 a sign of wrong usage. In many cases I would say yes, but in other methods I allow it intentionally (to not force the user to perform non-trivial checks before calling a method) and I prefer to keep all interfaces aligned. Utilization is computed for a specific target_criterion, so 0 may be considered valid, as long as the interface is defined as such.

---

Ok, I don't like it but it seems like this is the best option we currently got. Combined with the warning message when setting final ru I think that it is good enough